torchx/schedulers/local_scheduler.py

```diff
@@ -763,7 +763,11 @@ def _submit_dryrun(
             request, lambda p: pprint.pformat(asdict(p), indent=2, width=80)
         )
 
-    def _get_gpu_device_count(self) -> int:
+    def _device_count(self) -> int:
+        # this method deliberately does not use ``torch.cuda.device_count()``
+        # to avoid taking a dependency on pytorch
+        # this make sit possible to avoid a BUCK dependency (internally at Meta)
+        # on //caffe2:torch which slows down builds of //torchx:* rules
```

Comment:
I stumbled upon pyNVML library https://pypi.org/project/nvidia-ml-py/ doesn't quite solve the buck issue but would be lighterweight than torch perhaps? Just something I was thinking about, this is fine

Reply:
yeah it'll be nice to use that python-API, I'm not sure if `nvidia-ml-py` has been imported as a 3rd party lib in BUCK (I'm sure it has but I can't say 100% for sure) and how it'll make things better/worse for torchx in terms of BUCK deps/buildtime etc. Perhaps this is something @kurman or @priyaramani can help us with.